kongswap_adaptor/src/agent/ic_cdk_agent.rs

```diff
@@ -28,17 +29,14 @@ impl AbstractAgent for CdkAgent {
         canister_id: impl Into<Principal> + Send,
         request: R,
     ) -> Result<R::Response, Self::Error> {
-        let args_raw = request.payload().map_err(CdkAgentError::CandidEncode)?;
+        let raw_args = request.payload().map_err(CdkAgentError::CandidEncode)?;
 
-        let response =
-            ic_cdk::api::call::call_raw(canister_id.into(), request.method(), args_raw, 0)
-                .await
-                .map_err(|(err_code, err_message)| {
-                    CdkAgentError::IcCdk(err_code as i32, err_message)
-                })?;
+        let call_response = ic_cdk::call::Call::bounded_wait(canister_id.into(), request.method())
+            .take_raw_args(raw_args)
+            .change_timeout(15 * 60) // A time out of 15 minutes for requests.
```

Comment:
Please lift this to a top-level constant 

```suggestion
            .change_timeout(EXTERNAL_CALL_TIMEOUT_SECONDS)
```